actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -497,6 +508,17 @@ def expiry_options(options)
           end
         end
 
+        def cookie_metadata(options)
+          if request.use_cookies_with_metadata
+            metadata = expiry_options(options)
+            metadata[:purpose] = options[:cookie_name]
+          else
+            metadata = {}
+          end
+
+          metadata
+        end
+
         def parse(name, data); data; end
```

Comment:
This needs to accept purpose as well. Consider just doing def parse(name, data, metadata) here and elsewhere.

Reply:
@kaspth Just to be clear: Do you want me to change ` def parse(name, signed_message, purpose: nil)` to  `def parse(name, signed_message, metadata)` for signed/encrypted cookie parsing and line no 522?

Reply:
I was thinking that. But since `:purpose` is the only extra key, it's just clearer as is. The `parse` I commented on here does need `def parse(name, data, purpose:)` as well.